### ISSUE TITLE
[Bug]: Error while updating drafts: assets directory should only contain

### ISSUE THREAD

**Issue by etienne-ditchlabs** [Shopify/cli#4418]
### Please confirm that you have:

- [x] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [x] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Extension

### Expected behavior

`shopify app dev` should allow `.js.liquid` and `.css.liquid` files in the `/extensions/.../assets` folder.

As per the documentation (https://shopify.dev/docs/storefronts/themes/architecture#assets), themes can support this.

### Actual behavior

When I run `shopify app dev` on my app theme extension, I get the following error:
`Error while updating drafts: assets directory should only contain .jpg, .jpeg, .js, .css, .png, .svg, .json, .wasm files`

![Image](https://github.com/user-attachments/assets/4015cdf7-099c-40a7-8fe4-abe7c0b2b13d)

### Verbose output

N/A

### Reproduction steps

1. Create an app theme extension
2. In your extensions/.../assets folder, add a `test.js.liquid` or `test.css.liquid` file
3. Run `shopify app dev`

### Operating System

Mac os 14.2.1

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.66.1

**Comment by shehrozafzal** [comment]
I am facing this error in CLI while running `shopify app dev ` `Error while updating drafts: blocks directory should only contain .liquid files`
 while blocks directory only have one file which came with default installation of extension.

**Comment by mmatila** [comment]
Ran into this issue as well. Very new to Liquid and was trying to follow the [documentation](https://shopify.dev/docs/storefronts/themes/architecture#assets) for referencing external `.css` files in my `/blocks/foo.liquid`

**Comment by mmatila** [comment]
@etienne-ditchlabs I will be needing them so we're in the same boat 🫡

**Comment by mgmanzella** [comment]
the docs you linked were for themes no theme app extensions and i was under the impression this error was using from shopify app dev. is that correct? if so, i think you're right that we can improve the theme app extensions doc to explicitly state we dont support liquid files in the asset directory

**Comment by etienne-ditchlabs** [comment]
Yes, this error occurs when I'm trying to push an app theme extension and not a theme. I was under the impression that it would be the same logic/rules.

**Comment by shehrozafzal** [comment]
In my case i do not have any external file in my asset directory or any non .liquid file in my blocks folder but still i get an error every time i run the shopify app dev (in short words on newly created)

**Comment by mgmanzella** [comment]
👋 hi folks, caught up with the team a bit more about this. 

@etienne-ditchlabs we will not be adding support for `js.liquid` or `css.liquid` in theme app extensions due to performance concerns (this is why the one file limit for each of these exists for themes, see [here](https://shopify.dev/docs/storefronts/themes/architecture#supporting-assets)). theme app extensions are meant to have more constraints compared to themes to preserve theme performance and allow multiple apps to exist without affecting site performance. so im closing this issue as won't do. 

@shehrozafzal i would open a separate issue, what you're experiencing sounds unrelated to the original poster's issue. be sure to include the `--verbose` flag when running CLI commands and share those logs

thank you both for reporting 🙏
